From adf723a9e9191ad18d1bdf53154d8b2d37eead65 Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@consol.de>
Date: Sun, 10 Nov 2024 19:45:43 +0100
Subject: [PATCH] fix leaking object.cache file references (fixes #462)

using fopen opens the object.cache a second time and naemon closes just one afterwards.
So with every write of the objects.cache file, one reference will be kept and prevents
the file from beeing deleted which finally fills the disk with deleted files.
Using fdopen solves the issue, since it just reuses the existing fd.
---
 src/naemon/objects.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/naemon/objects.c b/src/naemon/objects.c
index 18f25605..e9a08cf7 100644
--- a/src/naemon/objects.c
+++ b/src/naemon/objects.c
@@ -42,7 +42,7 @@ int fcache_objects(char *cache_file)
 	}
 
 	/* open the cache file for writing */
-	fp = (FILE *)fopen(tmp_file, "w");
+	fp = (FILE *)fdopen(fd, "w");
 	if (fp == NULL) {
 		unlink(tmp_file);
 		nm_log(NSLOG_CONFIG_WARNING, "Warning: Could not open object cache data file '%s' for writing!\n", tmp_file);
